analyzer.go

```diff
@@ -362,10 +353,12 @@ func (gosec *Analyzer) load(pkgPath string, conf *packages.Config) ([]*packages.
 		}
 	}
 
-	// step 3/3 remove build tags from conf to proceed build correctly.
-	gosec.mu.Lock()
-	conf.BuildFlags = nil
-	defer gosec.mu.Unlock()
+	// step 2/2: pass in cli encoded build flags to build correctly.
+	conf := &packages.Config{
+		Mode:       LoadMode,
+		BuildFlags: CLIBuildTags(buildTags),
```

Comment:
Ins't this the only fix required to get properly the build flags propagated? Why's the reason for the extended refactoring?

Reply:
I think I mentioned in the PR description, I could move the config back, but make load take in both `buildTags` and the fixed `*packages.Config` which could reduce churn..? Was an attempt to stop potential top level config mutation due to being passed as a pointer.

* The default builder wants build _tags_ in the form `[]string{"tag1", "tag2"}`
* `packages.Load` wants cli build _flags_ in the form of `[]string{"-tags=tag1,tag2", "-ldflags=\"-s -w\""}`
* See mutex message above about the previous code mutating the top level config, potentially removing the build flags for the next worker.